com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/SerializedFrameSettings.cs

```diff
@@ -58,17 +58,17 @@ public bool HaveMultipleOverride(FrameSettingsField field)
                 return ref (obj as HDAdditionalCameraData).renderingPathCustomFrameSettings;
             if (obj is HDProbe)
                 return ref (obj as HDProbe).frameSettings;
-            if (obj is HDRenderPipelineAsset)
-                switch (HDRenderPipelineUI.selectedFrameSettings)
+            if (obj is HDDefaultSettings)
+                switch (DefaultSettingsPanelIMGUI.selectedFrameSettings)
                 {
-                    case HDRenderPipelineUI.SelectedFrameSettings.Camera:
-                        return ref (obj as HDRenderPipelineAsset).GetDefaultFrameSettings(FrameSettingsRenderType.Camera);
-                    case HDRenderPipelineUI.SelectedFrameSettings.BakedOrCustomReflection:
-                        return ref (obj as HDRenderPipelineAsset).GetDefaultFrameSettings(FrameSettingsRenderType.CustomOrBakedReflection);
-                    case HDRenderPipelineUI.SelectedFrameSettings.RealtimeReflection:
-                        return ref (obj as HDRenderPipelineAsset).GetDefaultFrameSettings(FrameSettingsRenderType.RealtimeReflection);
+                    case DefaultSettingsPanelIMGUI.SelectedFrameSettings.Camera:
+                        return ref (obj as HDDefaultSettings).GetDefaultFrameSettings(FrameSettingsRenderType.Camera);
+                    case DefaultSettingsPanelIMGUI.SelectedFrameSettings.BakedOrCustomReflection:
+                        return ref (obj as HDDefaultSettings).GetDefaultFrameSettings(FrameSettingsRenderType.CustomOrBakedReflection);
+                    case DefaultSettingsPanelIMGUI.SelectedFrameSettings.RealtimeReflection:
+                        return ref (obj as HDDefaultSettings).GetDefaultFrameSettings(FrameSettingsRenderType.RealtimeReflection);
                     default:
-                        throw new System.ArgumentException("Unknown kind of HDRenderPipelineUI.SelectedFrameSettings");
+                        throw new System.ArgumentException("Unknown kind of DefaultSettingsPanelIMGUI.SelectedFrameSettings");
```

Comment:
You can use [pattern matching](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/is) and avoid a second cast of obj to `HDDefaultSettings`

Also, I know is not coming from your changes but I think in this case a `switch`  will improve readability

```
ref FrameSettings GetData(Object obj)
        {
            switch (obj)
            {
                case HDAdditionalCameraData data:
                    return ref data.renderingPathCustomFrameSettings;
                case HDProbe probe:
                    return ref probe.frameSettings;
                default:
                {
                    if (obj is HDDefaultSettings defaultSettings)
                    {
                        switch (DefaultSettingsPanelIMGUI.selectedFrameSettings)
                        {
                            case DefaultSettingsPanelIMGUI.SelectedFrameSettings.Camera:
                                return ref defaultSettings.GetDefaultFrameSettings(FrameSettingsRenderType.Camera);
                            case DefaultSettingsPanelIMGUI.SelectedFrameSettings.BakedOrCustomReflection:
                                return ref defaultSettings.GetDefaultFrameSettings(FrameSettingsRenderType.CustomOrBakedReflection);
                            case DefaultSettingsPanelIMGUI.SelectedFrameSettings.RealtimeReflection:
                                return ref defaultSettings.GetDefaultFrameSettings(FrameSettingsRenderType.RealtimeReflection);
                        }
                    }

                    break;
                }
            }
            
            throw new System.ArgumentException("Unknown kind of DefaultSettingsPanelIMGUI.SelectedFrameSettings");
        }
```